c2corg_api/scripts/migration/documents/books.py

```diff
@@ -0,0 +1,110 @@
+from c2corg_api.models.book import Book, ArchiveBook, BOOK_TYPE
+from c2corg_api.models.document import DocumentLocale, ArchiveDocumentLocale, \
+    DOCUMENT_TYPE
+from c2corg_api.scripts.migration.documents.document import MigrateDocuments
+from c2corg_api.scripts.migration.documents.routes import MigrateRoutes
+
+
+class MigrateBooks(MigrateDocuments):
+
+    def get_name(self):
+        return 'books'
+
+    def get_model_document(self, locales):
+        return DocumentLocale if locales else Book
+
+    def get_model_archive_document(self, locales):
+        return ArchiveDocumentLocale if locales else ArchiveBook
+
+    def get_document_geometry(self, document_in, version):
+        return dict(
+            document_id=document_in.id,
+            id=document_in.id,
+            version=version,
+            geom=document_in.geom
+        )
+
+    def get_count_query(self):
+        return (
+            ' select count(*) '
+            ' from app_books_archives ba join books b on ba.id = b.id '
+            ' where '
+            ' b.redirects_to is null; '
+        )
+
+    def get_query(self):
+        return (
+            ' select '
+            '   ba.id, ba.document_archive_id, ba.is_latest_version, '
+            '   ba.is_protected, ba.redirects_to, '
+            '   ST_Force2D(ST_SetSRID(ba.geom, 3857)) geom, '
+            '   ba.activities, ba.book_types, '
+            '   ba.author, ba.editor, ba.url, ba.isbn, ba.langs, '
+            '   ba.nb_pages, ba.publication_date '
+            ' from app_books_archives ba join books b on ba.id = b.id '
+            ' where b.redirects_to is null '
+            ' order by ba.id, ba.document_archive_id;'
+        )
+
+    def get_count_query_locales(self):
+        return (
+            ' select count(*) '
+            ' from app_books_i18n_archives ba '
+            '   join books b on ba.id = b.id '
+            ' where b.redirects_to is null;'
+        )
+
+    def get_query_locales(self):
+        return (
+            ' select '
+            '    ba.id, ba.document_i18n_archive_id, ba.is_latest_version, '
+            '    ba.culture, ba.name, ba.description '
+            ' from app_books_i18n_archives ba '
+            '   join books b on ba.id = b.id '
+            ' where b.redirects_to is null '
+            ' order by ba.id, ba.culture, ba.document_i18n_archive_id;'
+        )
+
+    def get_document(self, document_in, version):
+        return dict(
+            document_id=document_in.id,
+            type=BOOK_TYPE,
+            version=version,
+
+            activities=self.convert_types(
+                document_in.activities, MigrateRoutes.activities),
+            book_types=self.convert_types(
+                document_in.book_types, MigrateBooks.book_types),
+            author=document_in.author,
+            editor=document_in.editor,
+            url=document_in.url,
+            isbn=document_in.isbn,
+            # langs=document_in.langs,
```

Comment:
When migrating `langs` we have to decide if we only accept languages defined in [default_langs](https://github.com/c2corg/v6_common/blob/master/c2corg_common/attributes.py#L1) or any language. E.g. "ot".

To get all used languages, you can run:
```
select lang from (select unnest(langs) lang from app_books_archives)t group by lang;
```